diff --git a/cmd/utils/flags.go b/cmd/utils/flags.go
--- a/cmd/utils/flags.go
+++ b/cmd/utils/flags.go
@@ -816,9 +816,9 @@
 	switch {
 	case ctx.IsSet(BootnodesFlag.Name) || ctx.IsSet(BootnodesV4Flag.Name):
 		if ctx.IsSet(BootnodesV4Flag.Name) {
-			urls = strings.Split(ctx.String(BootnodesV4Flag.Name), ",")
+			urls = SplitAndTrim(ctx.String(BootnodesV4Flag.Name))
 		} else {
-			urls = strings.Split(ctx.String(BootnodesFlag.Name), ",")
+			urls = SplitAndTrim(ctx.String(BootnodesFlag.Name))
 		}
 	case ctx.Bool(TestnetFlag.Name):
 		urls = params.TestnetBootnodes
